views/includes/inputs/date/identity-date-input.html

```diff
@@ -1,5 +1,55 @@
+{% set errorMess = '' %}
+{% set isIdentityDateDayError = false %}
+{% set isIdentityDateMonthError = false %}
+{% set isIdentityDateYearError = false %}
+
+{% if errors %}
+  {% for key, value in errors %}
+
+    {% if "identity_date" == key|string %}
+      {% set errorMess = value %}
+      {% if identity_date %}
+        {% if not identity_date["identity_date-day"] %}
+          {% set isIdentityDateDayError = true %}
+        {% endif %}
+        {% if not identity_date["identity_date-month"] %}
+          {% set isIdentityDateMonthError = true %}
+        {% endif %}
+        {% if not identity_date["identity_date-year"] %}
+          {% set isIdentityDateYearError = true %}
+        {% endif %}
+      {% endif %}
+    {% elif not errorMess and "identity_date" in key|string %}
+      {% set errorMess = value %}
+      {% if "-day" in key|string %}
+        {% set isIdentityDateDayError = true %}
+      {% elif "-month" in key|string %} 
+        {% set isIdentityDateMonthError = true %}
+      {% elif "-year" in key|string %} 
+        {% set isIdentityDateYearError = true %}
+      {% endif %}
+
+    {% endif %}
+  {% endfor %}    
+{% endif %}
+
+
+{% set identity_date_classes_year = "govuk-input--width-4 govuk-input" %}
+{% set identity_date_classes_month = "govuk-input--width-2 govuk-input" %}
+{% set identity_date_classes_day = "govuk-input--width-2 govuk-input" %}
+
+{% if isIdentityDateDayError %}
+  {% set identity_date_classes_day = identity_date_classes_day + "--error" %}
+{% endif %}
+{% if isIdentityDateMonthError %}
+  {% set identity_date_classes_month = identity_date_classes_month + "--error" %}
+{% endif %}
+{% if isIdentityDateYearError %}
+  {% set identity_date_classes_year = identity_date_classes_year + "--error" %}
+{% endif %}
+
 {{ govukDateInput({
-  errorMessage: errors.identity_date if errors,
+  errorMessage: errorMess,
```

Comment:
Could we call this errorMsg?

Reply:
went with identityDateErrorMsg so it doesn't clash with any other vars when included in templates